File: frontend/src/types/rawDocument.ts
export interface RawDocument {
  id: string;
  source: string;
  source_type: string;
  url: string | null;
  title: string;
  content: string;
  language: string;
  published_at: string | null;
  crawled_at: string | null;
  first_seen_at: string;
  content_hash: string;
  duplicate_group_id: string | null;
  credibility_score: number;
  metadata_json: Record<string, unknown> | null;
  created_at: string;
}

export interface RawDocumentCreate {
  source: string;
  source_type?: string;
  url?: string;
  title: string;
  content: string;
  language?: string;
  published_at?: string;
}

export interface RawDocumentListParams {
  offset?: number;
  limit?: number;
  source?: string;
  date_from?: string;
  date_to?: string;
}

File: backend/app/schemas/raw_document.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class RawDocumentCreate(BaseModel):
    source: str
    source_type: str = "manual"
    url: str | None = None
    title: str
    content: str
    language: str = "zh"
    published_at: datetime | None = None


class RawDocumentResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    source: str
    source_type: str
    url: str | None
    title: str
    content: str
    language: str
    published_at: datetime | None
    crawled_at: datetime | None
    first_seen_at: datetime | None = None
    created_at: datetime
    content_hash: str
    duplicate_group_id: UUID | None = None
    credibility_score: float


class RawDocumentListParams(BaseModel):
    offset: int = Field(default=0, ge=0)
    limit: int = Field(default=20, ge=1, le=100)
    source: str | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None